plugins/wasmedge_image/CMakeLists.txt

```diff
@@ -0,0 +1,142 @@
+  find_library(JPEG_STATIC NAMES libjpeg.a)
+  find_library(PNG_STATIC NAMES libpng16.a)
+  set_target_properties(wasmedgePluginWasmEdgeImageJPEG
+    PROPERTIES
+    IMPORTED_LOCATION ${JPEG_STATIC}
+    INTERFACE_INCLUDE_DIRECTORIES "${JPEG_INCLUDE_DIR}"
+  )
+  set_target_properties(wasmedgePluginWasmEdgeImagePNG
+    PROPERTIES
+    IMPORTED_LOCATION ${PNG_STATIC}
+    INTERFACE_INCLUDE_DIRECTORIES "${PNG_INCLUDE_DIR}"
+  )
+elseif(UNIX)
+  # Fetch and build libjpeg and libpng.
+  include(FetchContent)
+  FetchContent_Declare(
+    wasmedge_image_libpng
+    URL "https://downloads.sourceforge.net/libpng/libpng-1.6.39.tar.gz"
+    URL_HASH "SHA256=af4fb7f260f839919e5958e5ab01a275d4fe436d45442a36ee62f73e5beb75ba"
+  )
+  FetchContent_GetProperties(wasmedge_image_libpng)
+  if(NOT wasmedge_image_libpng_POPULATED)
+    message(STATUS "Downloading libpng source")
+    FetchContent_Populate(wasmedge_image_libpng)
+    message(STATUS "Downloading libpng source - done")
+  endif()
+
+  FetchContent_Declare(
+    wasmedge_image_libjpeg
+    URL "http://ijg.org/files/jpegsrc.v9e.tar.gz"
+    URL_HASH "SHA256=4077d6a6a75aeb01884f708919d25934c93305e49f7e3f36db9129320e6f4f3d"
+  )
+  FetchContent_GetProperties(wasmedge_image_libjpeg)
+  if(NOT wasmedge_image_libjpeg_POPULATED)
+    message(STATUS "Downloading libjpeg source")
+    FetchContent_Populate(wasmedge_image_libjpeg)
+    message(STATUS "Downloading libjpeg source - done")
+  endif()
+
+  add_custom_command(
+    OUTPUT ${wasmedge_image_libjpeg_SOURCE_DIR}/.libs/libjpeg.a
+    COMMAND ${CMAKE_COMMAND} -E env CFLAGS=-fPIC ./configure --enable-shared=off
+    COMMAND make
+    WORKING_DIRECTORY ${wasmedge_image_libjpeg_SOURCE_DIR}
+  )
+  add_custom_command(
+    OUTPUT ${wasmedge_image_libpng_SOURCE_DIR}/.libs/libpng16.a
+    COMMAND ${CMAKE_COMMAND} -E env CFLAGS=-fPIC ./configure --enable-shared=off
+    COMMAND make
+    WORKING_DIRECTORY ${wasmedge_image_libpng_SOURCE_DIR}
+  )
+  add_custom_target(wasmedgePluginWasmEdgeImageJPEG_target
+      ALL DEPENDS
+      ${wasmedge_image_libjpeg_SOURCE_DIR}/.libs/libjpeg.a
+  )
+  add_custom_target(wasmedgePluginWasmEdgeImagePNG_target
+      ALL DEPENDS
+      ${wasmedge_image_libpng_SOURCE_DIR}/.libs/libpng16.a
+  )
+  add_dependencies(wasmedgePluginWasmEdgeImageJPEG wasmedgePluginWasmEdgeImageJPEG_target)
+  add_dependencies(wasmedgePluginWasmEdgeImagePNG wasmedgePluginWasmEdgeImagePNG_target)
+  set_target_properties(wasmedgePluginWasmEdgeImageJPEG
+    PROPERTIES
+    IMPORTED_LOCATION ${wasmedge_image_libjpeg_SOURCE_DIR}/.libs/libjpeg.a
+    INTERFACE_INCLUDE_DIRECTORIES ${wasmedge_image_libjpeg_SOURCE_DIR}
+  )
+  set_target_properties(wasmedgePluginWasmEdgeImagePNG
+    PROPERTIES
+    IMPORTED_LOCATION ${wasmedge_image_libpng_SOURCE_DIR}/.libs/libpng16.a
+    INTERFACE_INCLUDE_DIRECTORIES ${wasmedge_image_libpng_SOURCE_DIR}
+  )
+endif()
+
+# Need zlib and boost.
+find_package(ZLIB REQUIRED)
+find_package(Boost 1.74.0)
```

Comment:
Do we need the version to be `1.74.0`?

Reply:
At least.
This is for the version >= 1.74.0.